src/statement-list/converters/expr2statement_list.cpp

```diff
@@ -25,3 +65,219 @@ std::string type2stl(const typet &type, const namespacet &ns)
+  {
+    std::vector<exprt> operands = expr.operands();
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, OR);
+  }
+}
+
+void expr2stlt::convert_xor(const exprt &expr)
+{
+  if(inside_bit_string)
+    convert_multiary_bool(expr.operands(), XOR);
+  else
+  {
+    std::vector<exprt> operands = expr.operands();
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, XOR);
+  }
+}
+
+void expr2stlt::convert_equal(const exprt &expr)
+{
+  std::vector<exprt> operands =
+    instrument_equal_operands(expr.op0(), expr.op1());
+  if(inside_bit_string)
+    convert_multiary_bool(operands, XOR);
+  else
+  {
+    convert_first_non_trivial_operand(operands);
+    convert_multiary_bool(operands, XOR);
+  }
+}
+
+void expr2stlt::convert_not(const exprt &expr)
+{
+  // If a NOT expression is being handled here it must always mark the
+  // beginning of a new bit string.
+  PRECONDITION(!inside_bit_string);
+
+  if(ID_symbol == expr.op0().id())
+  {
+    // Use AN to load the symbol.
+    is_reference = true;
+    result << AND << NOT_POSTFIX << OPERAND_SEPARATOR;
+    convert(expr.op0());
+    result << LINE_SEPARATOR;
+  }
+  else
+  {
+    // Use NOT to negate the RLO after the wrapped expression was loaded.
+    convert(expr.op0());
+    result << NOT LINE_SEPARATOR;
+  }
+}
+
+void expr2stlt::convert_symbol(const exprt &expr)
+{
+  if(is_reference)
+  {
+    result << REFERENCE_FLAG;
+    is_reference = false;
+  }
+  result << id2string(id_shorthand(expr.get(ID_identifier)));
+}
+
+void expr2stlt::convert_multiary_bool(
+  const std::vector<exprt> &operands,
+  const char operation)
+{
+  for(const exprt &op : operands)
+  {
+    if(op.id() == ID_not)
+    {
+      result << operation << NOT_POSTFIX;
+      convert_bool_operand(op.op0());
+    }
+    else
+    {
+      result << operation;
+      convert_bool_operand(op);
+    }
+  }
+}
+
+void expr2stlt::convert_bool_operand(const exprt &op)
+{
+  if(op.id() == ID_symbol)
+  {
+    is_reference = true;
+    result << OPERAND_SEPARATOR;
+    convert(op);
+    result << LINE_SEPARATOR;
+  }
+  else
+  {
+    inside_bit_string = false;
+    result << NESTING_OPEN_LINE_SEPARATOR;
+    convert(op);
+    result << NESTING_CLOSED_LINE_SEPARATOR;
+  }
+}
+
+void expr2stlt::convert_first_non_trivial_operand(std::vector<exprt> &operands)
+{
+  exprt non_trivial_op;
+  for(auto it(begin(operands)); it != end(operands); ++it)
+  {
+    if(
+      (ID_symbol == it->id()) ||
+      (ID_not == it->id() && ID_symbol == it->op0().id()))
+      continue;
+    else
+    {
+      non_trivial_op = *it;
+      operands.erase(it);
+      break;
+    }
+  }
+  // Important for some scenarios: Convert complex op first, set bit string
+  // flag to true afterwards.
+  if(non_trivial_op.is_not_nil())
+    convert(non_trivial_op);
+
+  inside_bit_string = true;
+}
+
+std::vector<exprt>
+expr2stlt::instrument_equal_operands(const exprt &lhs, const exprt &rhs)
```

Comment:
This can be a `static` function in the compilation unit, i.e. does not need to be in the header. Please check all functions whether they need member access and make them `static` otherwise.